src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -679,7 +679,7 @@ public DetailAST findField(String name) {
          * @return method calls of this frame.
          */
         public Set<DetailAST> getMethodCalls() {
-            return Collections.unmodifiableSet(methodCalls);
+            return new HashSet<>(methodCalls);
```

Comment:
this is also bad side effect of pitest.
is there way to explain pitst to not mutate `Collections.unmodifiableSet` ? 


Lets start to make list of bad side effects that following pitest case cause.
@Kevin222004 , please create issue in our issue tracker and we will link there such examples , you will make article at the end of Summer on what user should be aware of when dealing with pitest.

Reply:
explaination :- mostly similar to https://github.com/checkstyle/checkstyle/pull/13126#issue-1735665746

> is there way to explain pitst to not mutate Collections.unmodifiableSet ?

I have tried a lot to avoid removing this mutation but I really don't find anything


-------------------

might pure unit testing help not sure may be 

in community bonding period while i was practicnig pure unit testing i have faced this type of similar mutation 
at 
https://github.com/checkstyle/checkstyle/blob/81448ea434cda883b748d3d2c77b2f58a8b22bbe/config/pitest-suppressions/pitest-api-suppressions.xml#L3-L19
for https://github.com/checkstyle/checkstyle/blob/81448ea434cda883b748d3d2c77b2f58a8b22bbe/src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java#L313-L325

to kill this i have create test like which is killing this mutation 
```
    @Test
    public void testUnmodifiableGetSingleLineComment() {
        FileContents cppComments = new FileContents(new FileText(new File("filename"),
                Arrays.asList("// comment "," A + B " ,"// line" )));

        Map<Integer, TextBlock> comments = cppComments.getSingleLineComments();
        try {
            comments.remove(1);
            fail("Unmodifiable map");
        } catch (UnsupportedOperationException e) {
            // expected
        }
    }

    @Test
    public void testUnmodifiableGetBlockComments() {
        FileContents clangComments = new FileContents(new FileText(new File("filename"),
                Arrays.asList("/* comment */","/* A + B */" ,"//  Single line" )));

        Map<Integer, List<TextBlock>> comments = clangComments.getBlockComments();
        try {
            comments.remove(1);
            fail("Unmodifiable map");
        } catch (UnsupportedOperationException e) {
            // expected
        }
    }
```
I have not pushed this changes but i can if you want to see

Reply:
please review documentation of pitest on how to instruct pitest to skip some methods. 
if nothing works to let pitest avoid mutation, we can ask this question in pitest github project.

Reply:
![image](https://github.com/checkstyle/checkstyle/assets/97679350/ab34627d-d94f-4a43-9d0b-9bde9c159baf)
https://docs.arcmutate.com/docs/exclusions.html

@romani here is the way to avoid the pitest

Reply:
Please show how it will looks like for both:
>By method name, By annotation

Preferably `By method name`

Reply:
@romani please check I have implemented it.